### ISSUE TITLE
VoiceOver accessibility issues

### ISSUE THREAD

**Issue by tazz4843** [ihateblueb/snowdrop#175]
**Description**
Given all the issues I've reported so far have been accessibility ones I figured I might as well try VoiceOver (iOS's screen reader) and see how it was.

I can navigate around the UI fine, but:

* Every single button is reported as just "Button". This can probably be fixed with [`accessibilityLabel`](https://developer.apple.com/documentation/uikit/uiaccessibilityelement/accessibilitylabel).
* Trying to activate boost or favourite buttons triggers no UI update until the UI is repainted (ie opening the post view or reopening the app) so it seems like boost/favourite do nothing.
* Trying to drag around the focus with single finger tap and drag begins scrolling extremely fast near the top or bottom edges. I suspect this is because it's focusing to the next major UI element, which is the next post. Native SwiftUI apps do not have this problem because they refuse scrolling when dragging around the focus.
* The entire UI becomes untouchable (action rejected sound) if I open any dropdown (timeline selector ie) and then try to navigate out (ie tap outside of the menu). Native UI elements say "Close context menu" when this is done instead. The return gesture (two finger scrub) does work to close the menu and return focus to the parent element so this isn't major.
* The tab selectors at the bottom straight up don't work. Trying to select the notification tab does absolutely nothing.
* Somehow the "post characters remaining" hint is read out as individual digits instead of as a full number (ie my instance's limit of 102,400 is read out as "one zero two four zero zero" instead of "one hundred two thousand four hundred")
* Navigating over emoji in the emoji picker requires moving to the next element twice to move one emoji. This is because the first object is read as "Button" which I imagine is "select this emoji", while the second object is slightly smaller than the first and reads the actual emoji name. I *believe* (not certain) this could be done by setting `accessibilityLabel` to "Insert emoji" and [`accessibilityValue`](https://developer.apple.com/documentation/uikit/uiaccessibilityelement/accessibilityvalue) to the emoji name.
* Settings do not have their current state read out when hovered. Their child element (ie the boolean toggle) is selectable (this is not how it's done in the settings app and most other settings UIs) but also does not read out the current state. In other words there's no indication of the current state at all to a user without sight. Ideally this would be fixed by following other apps and making activating the label toggle the element and read state, instead of having a child checkbox element at all. This is probably doable with [`accessibilityTraits`](https://developer.apple.com/documentation/uikit/uiaccessibilityelement/accessibilitytraits). This works fine with settings elements that are dropdowns (the default post visiblity).
* Focusing a photo without alt text just reads out "Button" instead of "Image".

I imagine most of these are upstream UI framework issues but I figured I would note these anyways. This is in my testing as a fully sighted user so I am absolutely certain I missed some.

If you need additional explanation or a video of any specific ones to understand let me know and I can do so.

**Environment**
 - Device: iPhone 15 Pro Max
 - OS: iOS 26.6
 - App version: 0.0.6-alpha (65b5010@master)

**Comment by ihateblueb** [comment]
screen reader support on every platform is pretty bad right now due to the lack of labels on most things, i indented to go back over and add stuff but that's a bad mentality so going forward all additions should have proper screen reader labels and in batches i'll go over old stuff.

**Comment by ihateblueb** [comment]
Relevant modifier: semantics with contentDescription
